src/parser/monk/mistweaver/modules/talents/InvokeChiJi.js

```diff
@@ -2,23 +2,72 @@ import React from 'react';
 import Statistic from 'interface/statistics/Statistic';
 import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
 import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
-import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
 import ItemHealingDone from 'interface/ItemHealingDone';
+import SpellLink from 'common/SpellLink';
+
+/** 
+ * Blackout Kick, Totm BoKs, Rising Sun Kick and Spinning Crane Kick generate stacks of Invoke Chi-Ji, the Red Crane, which reduce the cast time and mana 
+ * cost of Enveloping Mist by 33% per stack, up to 3 stacks.
+ * These abilities also heal 2 nearby allies for a Gust of Mist heal.
+ * Casting Enveloping Mist while Chiji is active applies Enveloping Breath on up to 6 nearby allies within 10 yards.
+*/
 class InvokeChiJi extends Analyzer {
+  chijiActive = false;
+  //healing breakdown vars
   gustHealing = 0;
   envelopHealing = 0;
-
+  //stack breakdown vars
+  chijiStackCount = 0;
+  castsBelowMaxStacks = 0;
+  wastedStacks = 0;
+  freeCasts = 0;
+  //missed GCDs vars
+  missedGlobals = 0;
+  chijiStart = 0;
+  chijiGlobals = 0;
+  chijiUses = 0;
+  lastGlobal = 0;
+  
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTalent(SPELLS.INVOKE_CHIJI_THE_RED_CRANE_TALENT.id);
     if (!this.active) {return;}
     this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.GUST_OF_MISTS_CHIJI), this.handleGust);
     this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.ENVELOPING_BREATH), this.handleEnvelopingBreath);
+    this.addEventListener(Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.INVOKE_CHIJI_THE_RED_CRANE_BUFF), this.handleApplyChijiStack);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.INVOKE_CHIJI_THE_RED_CRANE_BUFF), this.handleApplyChijiStack);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.ENVELOPING_MIST), this.handleEnvelopCast);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.INVOKE_CHIJI_THE_RED_CRANE_TALENT), this.handleChijiStart);
+    this.addEventListener(Events.GlobalCooldown.by(SELECTED_PLAYER), this.handleGlobal);
+  }
+
+  //missed gcd mangement
+  handleChijiStart(event) {
+    this.chijiActive = true;
+    this.chijiStart = this.lastGlobal = event.timestamp;
+    this.chijiGlobals += 1;
+    this.chijiUses +=1;
+  }
+
+  handleGlobal(event){
+    //cant get eventlistener to find chiji's death so we will check against duration here
+    if(event.timestamp - this.chijiStart > SPELLS.INVOKE_CHIJI_THE_RED_CRANE_TALENT.duration) {
+      this.chijiActive = false;
+    }
+    if(this.chijiActive) {
+      this.chijiGlobals += 1;
+      //if timebetween globals is longer than the gcd add the difference to the missed gcd tally
+      //we dont care about accounting for channels/casts longer than the gcd since you shouldnt be using them during chiji anyway
```

Comment:
If you're taking kyrian it might actually be worth to channel essence font during the middle of chi-ji so the time between should be accounted for imo
these could be handled with beginchannel and endchannel events

Reply:
added handling for kyrian chiji